tests/unit/util/pack_traversal.cpp

```diff
@@ -0,0 +1,577 @@
+//  Copyright (c) 2017 Denis Blank
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/lcos/future.hpp>
+#include <hpx/traits/future_traits.hpp>
+#include <hpx/traits/is_future.hpp>
+#include <hpx/util/lightweight_test.hpp>
+#include <hpx/util/pack_traversal.hpp>
+#include <algorithm>
+#include <list>
+#include <memory>
+#include <vector>
+
+using namespace hpx;
+using namespace hpx::util;
+using namespace hpx::util::detail;
+
+struct all_map_float
+{
+    template <typename T>
+    float operator()(T el) const
+    {
+        return float(el + 1.f);
+    }
+};
+
+struct my_mapper
+{
+    template <typename T,
+        typename std::enable_if<std::is_same<T, int>::value>::type* = nullptr>
```

Comment:
I see, so rejecting then means to discard the input value? I would have expected a compile failure in case of a missing overload. What's the rationale here?